### ISSUE TITLE
Overzealous enforcement of "no spaces" in text of gloss

### ISSUE THREAD

**Issue by cneidle** [DCS-LCSR/SignStream3#677]
Spaces must be allowed within quotation marks -- i.e., for gestures and classifiers.

<img width="1097" alt="Screen Shot 2023-04-12 at 2 47 18 PM" src="https://user-images.githubusercontent.com/13629122/231555869-5e06a20a-2b2c-4de8-9872-f5823f73db39.png">

<img width="491" alt="Screen Shot 2023-04-12 at 2 46 53 PM" src="https://user-images.githubusercontent.com/13629122/231555877-9daadf8c-4619-41ed-9f98-d6ab9428b256.png">

**Comment by cneidle** [comment]
I was able to get around this by inserting a non-breaking space, but I don't know if that is going to cause a problem later on, for Augustine's parser... 

copying Augustine @kwasiopoku

**Comment by gregorydimitriadis** [comment]
This wound up being a relatively quick fix.

**Comment by cneidle** [comment]
Excellent!

thanks 

Carol

**Comment by cneidle** [comment]
Listed as "code complete" as of May 23, but it is still a problem...

**Comment by cneidle** [comment]
Note that non-breaking spaces no longer work.  FWIW.

**Comment by gregorydimitriadis** [comment]
Main issue remaining here: allowing spaces between quotes, and not elsewhere.

**Comment by gregorydimitriadis** [comment]
Carol, how should this apply when creating new glosses? When we create the text values (before providing a start/end time) would this same logic of allowing spaces only between quotes apply there as well? As in, spaces not between quotes would still separate into different 'words' but spaces inside quotes would be considered part of the same 'word'

**Comment by cneidle** [comment]
Yes.  That sounds right.  Thanks!

**Comment by gregorydimitriadis** [comment]
@cneidle  Does this apply to allowing spaces within double quotes, or within single quotes as well?

**Comment by cneidle** [comment]
Best to also allow spaces within single quotes as well. Thanks!Sent from my iPhone; please excuse typos.On Oct 9, 2023, at 9:44 AM, Greg Dimitriadis ***@***.***> wrote:﻿
@cneidle  Does this apply to allowing spaces within double quotes, or within single quotes as well?

**Comment by gregorydimitriadis** [comment]
@cneidle are single quotes ever used in the names of glosses and/or gestures? I'm wondering if we can just disallow single quotes from being entered here, which may make some things simpler on the backend. Or if that would be too heavy-handed, and we would rather allow single quotes.

**Comment by cneidle** [comment]
Hi Greg,

I don’t believe we need to allow single quotes in the same context
where we need to allow double quotes.

However, we do need to allow ‘ in a gloss, between double quotes,
e.g., (1h)5"that's the way it is” or (1h)5"what's going on?"

Thanks,
  Carol

**Comment by cneidle** [comment]
This seems to work, in allowing, e.g., 5"test this item".  We discussed single quotes, and I thought you decided not to allow that.  But it is allowed, in fact -- which is not a problem !

So I think we can close this issue :-).   Thank you.

**Comment by gregorydimitriadis** [comment]
Yes, single quotes are allowed within double quotes, like the examples here. It also works in reverse, so:

`word'words inside"single quote'`

would also be valid (despite not making logical sense).

**Comment by cneidle** [comment]
Close this issue now?  Thanks very much.

**Comment by cneidle** [comment]
There are still problems
<img width="1079" alt="Screen Shot 2024-03-14 at 3 40 10 PM" src="https://github.com/DCS-LCSR/SignStream3/assets/13629122/bd4c4023-c748-4b0d-b354-2f11696acfd7">

**Comment by cneidle** [comment]
Furthermore:  

<img width="1054" alt="Screen Shot 2024-03-14 at 3 43 36 PM" src="https://github.com/DCS-LCSR/SignStream3/assets/13629122/89d640d9-ff57-42a4-bdd8-e1131aee611a">

**Comment by cneidle** [comment]
The above two reports involve v. 3.4.1. OS 11.7.10

**Comment by cneidle** [comment]
This mostly seems to have been fixed in 3.4.1.   The only time this error shows up is when editing a pre-existing gloss that contains quotation marks.  This can be resolved by retyping the quotation marks.  (I assume there is some issue with legacy special characters used for " .)   If this remains as it is, we should probably include a warning in the documentation for the next release.

**Comment by cneidle** [comment]
Would it be possible to add whatever character(s) SignStream thinks the quotation marks from previous versions might be to the set of characters allowed to have spaces in between them?
